Source/JavaScriptCore/dfg/DFGGraph.h

```diff
@@ -816,112 +830,79 @@ class Graph final : public virtual Scannable {
     }
 
     bool isWatchingNumberToStringWatchpoint(Node* node)
     {
-        if (m_plan.isUnlinked())
-            return false;
-
         JSGlobalObject* globalObject = globalObjectFor(node->origin.semantic);
         InlineWatchpointSet& set = globalObject->numberToStringWatchpointSet();
-        return isWatchingGlobalObjectWatchpoint(globalObject, set);
+        return isWatchingGlobalObjectWatchpoint(globalObject, set, LinkerIR::Type::NumberToStringWatchpointSet);
     }
 
     bool isWatchingStructureCacheClearedWatchpoint(Node* node)
     {
-        if (m_plan.isUnlinked())
-            return false;
-
         JSGlobalObject* globalObject = globalObjectFor(node->origin.semantic);
         InlineWatchpointSet& set = globalObject->structureCacheClearedWatchpointSet();
-        return isWatchingGlobalObjectWatchpoint(globalObject, set);
+        return isWatchingGlobalObjectWatchpoint(globalObject, set, LinkerIR::Type::StructureCacheClearedWatchpointSet);
     }
 
     bool isWatchingStringSymbolReplaceWatchpoint(Node* node)
     {
-        if (m_plan.isUnlinked())
-            return false;
-
         JSGlobalObject* globalObject = globalObjectFor(node->origin.semantic);
         InlineWatchpointSet& set = globalObject->stringSymbolReplaceWatchpointSet();
-        return isWatchingGlobalObjectWatchpoint(globalObject, set);
+        return isWatchingGlobalObjectWatchpoint(globalObject, set, LinkerIR::Type::StringSymbolReplaceWatchpointSet);
     }
 
     bool isWatchingRegExpPrimordialPropertiesWatchpoint(Node* node)
     {
-        if (m_plan.isUnlinked())
-            return false;
-
         JSGlobalObject* globalObject = globalObjectFor(node->origin.semantic);
         InlineWatchpointSet& set = globalObject->regExpPrimordialPropertiesWatchpointSet();
-        return isWatchingGlobalObjectWatchpoint(globalObject, set);
+        return isWatchingGlobalObjectWatchpoint(globalObject, set, LinkerIR::Type::RegExpPrimordialPropertiesWatchpointSet);
     }
 
     bool isWatchingArraySpeciesWatchpoint(Node* node)
     {
-        if (m_plan.isUnlinked())
-            return false;
-
         JSGlobalObject* globalObject = globalObjectFor(node->origin.semantic);
         InlineWatchpointSet& set = globalObject->arraySpeciesWatchpointSet();
-        return isWatchingGlobalObjectWatchpoint(globalObject, set);
+        return isWatchingGlobalObjectWatchpoint(globalObject, set, LinkerIR::Type::ArraySpeciesWatchpointSet);
     }
 
     bool isWatchingArrayPrototypeIsSaneChainWatchpoint(Node* node)
     {
-        if (m_plan.isUnlinked())
-            return false;
-
         JSGlobalObject* globalObject = globalObjectFor(node->origin.semantic);
         InlineWatchpointSet& set = globalObject->arrayPrototypeChainIsSaneWatchpointSet();
-        return isWatchingGlobalObjectWatchpoint(globalObject, set);
+        return isWatchingGlobalObjectWatchpoint(globalObject, set, LinkerIR::Type::ArrayPrototypeIsSaneChainWatchpointSet);
```

Comment:
Why name this `ArrayPrototypeIsSaneChainWatchpointSet` instead of `ArrayPrototypeChainIsSaneWatchpointSet` to match `arrayPrototypeChainIsSaneWatchpointSet()`?

Reply:
Changed